### DISCUSSION TITLE
Is there a way to make Vercel serverless functions readable only to my site?

### DISCUSSION THREAD

**Discussion by DeveloperKhan** [vercel/community#2239] [Help]
### Summary

Hello, I have a Gatsby site deployed on the Vercel hobby plan. I recently added serverless functions for it, now I can call the endpoints from `my-site-url/api`.

I want to restrict this usage to only calls made within my site (or my localhost if using `vercel dev`). Is there an out of the box solution for this?

### Example

dracoviz-site-bdg42vypx-pogodracoviz-gmailcom.vercel.app/api

**Comment by maddsua** [comment]
Generally speaking, a browser will refuse to complete the request to your API if it is called from a different origin.
I'll give you an example of how it work on Netlify, should be applicable to Vercel too.

So, you have a function on a site A. You can request it without any issues from the site A. But if a site B requests it, the function should properly respond to a CORS Pre-flight first, and only then it would be possible to transfer any data. This is the default browser behavior at the moment.
But if a server C requests a function from the site A, nothing will stop it, and you can easily do the same from your local machine using curl.

The only solution is to always check the `Origin` header, and if it does not match your site, just drop the request. This will 100% work for browsers, but it only kinda works for the requests from other servers as they simply can fake the origin header.

I have a production example of a code like that, let me know if you wanna have a look
